resharper/resharper-unity/src/Unity/Core/Application/UI/Help/ShowUnityHelp.cs

```diff
@@ -62,10 +64,22 @@ public static string FormatDocumentationKeyword(string keyword)
         [NotNull]
         public Uri GetUri([NotNull] string keyword)
         {
+            // VS specifies keyword as SerializeField.#ctor
+            // Rider specifies keyword as SerializeField.-ctor
+            // result for offline doc should be SerializeField-ctor/SerializeField, depending on presence of specific doc on the disk
+            // result for online doc should contain dot, otherwise nothing is found
+
             var documentationRoot = GetDocumentationRoot();
-            return GetFileUri(documentationRoot, $"ScriptReference/{keyword}.html")
-                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.Replace('.', '-')}.html")
+            var res = GetFileUri(documentationRoot, $"ScriptReference/{keyword}.html")
+                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.ReplaceLast('.', '-')}.html") // properties
+                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.Replace(".-", "-")}.html") // ctor in Rider
+                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.Replace(".-ctor", "")}.html") // ctor in Rider doesn't exist, so goto type doc
+                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.Replace(".#", "-")}.html") // ctor in VS
+                   ?? GetFileUri(documentationRoot, $"ScriptReference/{keyword.Replace(".#ctor", "")}.html") // ctor in VS doesn't exist, so goto type doc
```

Comment:
This is confusing. Can we do some of these transformations earlier? We know that these patterns won't ever match doc files, so why bother trying? If we always replace `.-` and `.#` with just `-`, then that will fix both Rider and VS style `ctor` links. And then we can have a fallback to replace `-ctor` to try to use the type.